dataflow/manual/content.tex

```diff
@@ -1734,6 +1734,34 @@ \section{Example: Live Variable}
 
 \flow{LiveSimple}{.33}{1}{Simple sequential program to illustrate live variable.  Intermediate analysis results are shown.}
 
+\section{Example: Very Busy Expression Analysis}
+
+Unlike live variable analysis, very busy expression analysis is a backward analysis that
+instead of combining abstract states from successors, intersects abstract states from successors.
+\\
+The main class is \code{org.checkerframework.dataflow.cfg.playground.BusyExpressionPlayground}.
+
+\textbf{Abstract values.} The \code{BusyExprValue} is a very busy expression
+ (which is represented by a node) wrapper turning node into
+abstract value. The node can be any \code{BinaryOperationNode}, such as \code{NumericalAdditionNode}
+ and \code{LeftShiftNode}.
+
+\textbf{The store.} The busy expression store \code{BusyExprStore} has a field
+\code{busyExprValueSet} which contains a set of \code{BusyExprValue}s.
+ If a node is \code{BinaryOperationNode}, \code{addUseInExpression(Node)}
+ will recursively analyze the rhs and lhs of the node to determine
+ if they are also \code{BinaryOperationNode}. We then use \code{putBusyExpr(BusyExprValue)}
+ to put the expression in the set. \code{killBusyExpr(Node)} is used to
+ kill all expressions in set containing the variable that is re-assigned.
+
+\textbf{The transfer function.} The transfer function \code{BusyExprTransfer}
+initializes empty stores at normal and exceptional exit blocks (because this
+is a backward transfer function). The transfer function visits assignment, method invocation,
+ return statement to update the information of busy expressions for each node in the stores.
```

Comment:
> for each node in the stores

I think it's updating values in the stores

Reply:
fixed it.